src/function_lua.c

```diff
@@ -241,17 +241,13 @@ static void luaEngineFreeFunction(engineCtx *engine_ctx,
 }
 
 static void luaRegisterFunctionArgsInitialize(compiledFunction *func,
-                                              char *name,
-                                              size_t name_len,
-                                              char *desc,
-                                              size_t desc_len,
+                                              sds name,
+                                              sds desc,
                                               luaFunctionCtx *lua_f_ctx,
                                               uint64_t flags) {
     *func = (compiledFunction){
-        .name = name,
-        .name_len = name_len,
-        .desc = desc,
-        .desc_len = desc_len,
+        .name = createStringObject(name, sdslen(name)),
```

Comment:
`luaGetStringSds` duplicates the string and `createStringObject` duplicates it again here.

We can rename/change `luaGetStringSds` to `luaGetStringObject` and avoid copying it twice. Then it's also more consistently passed around as an jrobj.

`luaGetStringSds` is actually only used for these name and desc, just three calls in this file, nowhere else in the code base.

?

Reply:
done